TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -857,9 +883,18 @@ private static bool HasChildrenThatDeclareData([NotNull] DataDefinition dataDefi
             {
                 var lastChild = dataDefinition.Children[dataDefinition.ChildrenCount - 1];
 
+                if (lastChild.CodeElement == null) //index
+                {
+                    return false;
```

Comment:
Please add a `Debug.Assert(lastChild is IndexDefinition)` before returning

Reply:
Done